components/massmint/Massmint.vue

```diff
@@ -156,10 +164,35 @@ const startMint = () => {
   mintModalOpen.value = true
   isMinting.value = true
 
-  // fake mint
-  setTimeout(() => {
-    isMinting.value = false
-  }, 5000)
+  const { isLoading, status, collectionUpdated, isError } = useMassMint(
+    Object.values(NFTS.value) as NFTToMint[],
+    selectedCollection.value as MintedCollection
+  )
+
+  watch(
+    [isLoading, status, collectionUpdated, isError],
+    ([isLoadingV, statusV], [isLoadingOldV]) => {
+      isMinting.value = isLoadingV
+      mintStatus.value = statusV
+
+      if (isLoadingOldV && !isLoadingV) {
+        mintModalOpen.value = false
+        if (!isError.value && statusV !== 'loader.sign') {
+          showNotification(
+            $i18n.t('massmint.continueToCollectionPage'),
+            notificationTypes.success
+          )
+        }
+      }
+
+      //redirect to collection page when collection is updated
+      if (collectionUpdated.value) {
+        router.push({
```

Comment:
[navigateTo](https://nuxt.com/docs/api/utils/navigate-to#navigateto)
```navigateTo(`/${urlPrefix.value}/collection/${selectedCollection.value?.id}`)```

Reply:
done